pkg/provider/server.go

```diff
@@ -29,9 +29,22 @@ type Attributes struct {
 	Application   string `json:"application,omitempty"`
 	Profile       string `json:"profile,omitempty"`
 	FileType      string `json:"fileType,omitempty"`
+	Raw           []Raw  `json:"raw"`
+}
+
+type Raw struct {
+	Source string `json:"source,omitempty"`
+	Target string `json:"target,omitempty"`
 }
 
 func (a *Attributes) verify() error {
+
+	for _, item := range a.Raw {
+		if item.Source == "" || item.Target == "" {
+			return fmt.Errorf("Source or Target not set")
```

Comment:
Nit: Adding the index to the error message might help us identify what is wrong faster in the future.
Example: `Source or Target of item raw[0] not set`